lib/spreadsheet_writer.rb

```diff
@@ -38,9 +39,13 @@ def session
     end
 
     def worksheet
-      @worksheet ||= session.spreadsheet_by_key(spreadsheet_id).worksheets[worksheet_pos]
+      @worksheet ||= use_gid ? spreadsheet.worksheet_by_gid(worksheet_pos) : spreadsheet.worksheets[worksheet_pos]
```

Comment:
👍 looks good, is there any case where we don't want to look up a worksheet by `gid`?

Reply:
Yeah, the other jobs that use SpreadsheetWriter currently lookup the worksheets by key. I'm going to slowly bring in the new stuff, but wanted to be backwardly compatible for now